apps/passport-server/test/devconnect.spec.ts

```diff
@@ -580,6 +584,59 @@ describe("devconnect functionality", function () {
     }
   );
 
+  step("should be able to log in with a device login", async function () {
+    const secret = mocker
+      .get()
+      .ordersByEventID.get(mocker.get().eventA.slug)
+      ?.find((order) => order.email == mocker.get().EMAIL_1)?.secret;
+
+    if (!secret) {
+      throw new Error("No secret found");
+    }
+
+    const fetchedDeviceLogin = await fetchDevconnectDeviceLoginTicket(
+      db,
+      mocker.get().EMAIL_1,
+      secret
+    );
+
+    expect(fetchedDeviceLogin).is.not.undefined;
+
+    const result = await testDeviceLogin(
+      application,
+      mocker.get().EMAIL_1,
+      secret
+    );
+
+    if (!result) {
+      throw new Error("Not able to login with device login");
+    }
+
+    expect(result.user).to.include({ email: mocker.get().EMAIL_1 });
+  });
+
+  step(
+    "should not be able to log in with a device login for non-superuser",
+    async function () {
+      const tickets = await fetchAllNonDeletedDevconnectPretixTickets(db);
+      const secret = tickets.find(
+        (ticket) => ticket.email == mocker.get().EMAIL_3
+      )?.secret;
+
+      if (!secret) {
+        throw new Error("No secret found");
+      }
+
+      const fetchedDeviceLogin = await fetchDevconnectDeviceLoginTicket(
+        db,
+        mocker.get().EMAIL_3,
+        secret
+      );
+
+      expect(fetchedDeviceLogin).is.undefined;
```

Comment:
this test isn't testing that you aren't able to log in in the case you aren't a superuser - it's testing some internal logic by which the server decides whether you should or shouldn't be logged in